File: Tareas/11.c
#include <stdio.h>
#include <math.h>
#include "../core/aritmetica.h"
#include "../metodos/solvers.h"

double f3(double x) { 
    return pow(x, 3.0) - 7.0 * pow(x, 2.0) + 14.0 * x - 6.0; 
}

int main() {
    printf("==============================================================\n");
    printf("   EJERCICIO 3: Biseccion (Tol: 10^-2)\n");
    printf("==============================================================\n");

    printf("\n--- Inciso A: Intervalo [0, 1] ---\n");
    biseccion(f3, 0.0, 1.0, 1e-2, 100, 6);

    printf("\n--- Inciso B: Intervalo [1, 3.2] ---\n");
    biseccion(f3, 1.0, 3.2, 1e-2, 100, 6);

    printf("\n--- Inciso C: Intervalo [3.2, 4] ---\n");
    biseccion(f3, 3.2, 4.0, 1e-2, 100, 6);

    return 0;
}

File: Tareas/11.py
import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from mpmath import mp
from metodos.solvers import biseccion, newton_raphson, secante, posicion_falsa, muller

mp.dps = 50
TOL = 1e-4
MAX_ITER = 100
DEC = 6

def f_7(x):
    return 600*x**4 - 550*x**3 + 200*x**2 - 20*x - 1

if __name__ == "__main__":
    print("\n=== EJERCICIO 7a: Bisección [0.1, 1.0] ===")
    biseccion(f_7, 0.1, 1.0, TOL, MAX_ITER, DEC)

    print("\n=== EJERCICIO 7b: Newton (x0 = 0.55) ===")
    newton_raphson(f_7, 0.55, TOL, MAX_ITER, DEC)

    print("\n=== EJERCICIO 7c: Secante (x0 = 0.1, x1 = 1.0) ===")
    secante(f_7, 0.1, 1.0, TOL, MAX_ITER, DEC)

    print("\n=== EJERCICIO 7d: Posición Falsa [0.1, 1.0] ===")
    posicion_falsa(f_7, 0.1, 1.0, TOL, MAX_ITER, DEC)

    print("\n=== EJERCICIO 7e: Müller (x0 = 0.1, x1 = 0.55, x2 = 1.0) ===")
    muller(f_7, 0.1, 0.55, 1.0, TOL, MAX_ITER, DEC)